cmd/cluster/context.go

```diff
@@ -601,28 +619,38 @@ func GetPagerdutyClient(usertoken string, oauthtoken string) (*pd.Client, error)
 	if client != nil {
 		return client, err
 	}
+
 	client, err = getPDOauthClient(oauthtoken)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create both user and oauth clients for pd")
+		return nil, fmt.Errorf("failed to create both user and oauth clients for pd, neither key pd_oauth_token or pd_user_token are set in config file")
 	}
 	return client, err
 }
 
-func GetPDSeviceID(baseDomain string, usertoken string, oauthtoken string) (string, error) {
+func GetPDSeviceID(baseDomain string, usertoken string, oauthtoken string, team_ids []string) ([]string, error) {
 	ctx := context.TODO()
 	pdClient, err := GetPagerdutyClient(usertoken, oauthtoken)
-	lsResponse, err := pdClient.ListServicesWithContext(ctx, pd.ListServiceOptions{Query: baseDomain})
+	if err != nil {
+		fmt.Printf("Failed to GetPagerdutyClient %q\n", err)
+		return []string{}, err
+	}
+
+	// Gets the PD Team IDS
+	teams := getPDTeamIDs(team_ids)
+
+	lsResponse, err := pdClient.ListServicesWithContext(ctx, pd.ListServiceOptions{Query: baseDomain, TeamIDs: teams})
```

Comment:
```suggestion
	lsResponse, err := pdClient.ListServicesWithContext(context.TODO(), pd.ListServiceOptions{Query: baseDomain, TeamIDs: teams})
```

No need to store this in a variable named `ctx` if we're just using it once. Let's delete line 631 as well where this variable is defined.